File: backend/Services/Database/AvailableDay.cs
namespace Services.Database
{
  public class AvailableDay
{
    public int Id { get; set; } 
    public int TutorScheduleId { get; set; } 
    public TutorSchedule TutorSchedule { get; set; } = null!;

    public DayOfWeek DayOfWeek { get; set; }
    public TimeSpan StartTime { get; set; }
    public TimeSpan EndTime { get; set; }
}

}

File: UI/justspeak_mobile/lib/models/available_day.dart
import 'package:json_annotation/json_annotation.dart';

part 'available_day.g.dart';

@JsonSerializable()
class AvailableDay {
  final int dayOfWeek; // 1=MON, 2=TUE, etc.
  final String? startTime; // format HH:mm:ss
  final String? endTime;   // format HH:mm:ss

  AvailableDay({
    required this.dayOfWeek,
    this.startTime,
    this.endTime,
  });

  factory AvailableDay.fromJson(Map<String, dynamic> json) => _$AvailableDayFromJson(json);
  Map<String, dynamic> toJson() => _$AvailableDayToJson(this);
}
